compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -446,9 +447,12 @@ object RefChecks {
           }
 
       def ignoreDeferred(member: SingleDenotation) =
-        member.isType ||
-          member.symbol.isSuperAccessor || // not yet synthesized
-          member.symbol.is(JavaDefined) && hasJavaErasedOverriding(member.symbol)
+        member.isType || {
+          val mbr = member.symbol
+          mbr.isSuperAccessor || // not yet synthesized
+          ShortcutImplicits.isImplicitShortcut(mbr) || // only synthesized when referenced
```

Comment:
The comment "only synthesized when referenced" is confusing, if I understand correctly direct method _trees_  are always synthesized.